### ISSUE TITLE
Support tmLanguage files natively

### ISSUE THREAD

**Issue by chtenb** [trishume/syntect#418]
I would like to make a feature request for support for tmLanguage files and/or the VSCode flavour of this. It is a very prevalent grammar format which would allow us to tap into a vastly bigger ecosystem directly. I'm aware that there are converters between the formats, however I've found that they don't work correctly in all cases. Moreover, it would save maintainers of grammars a lot of hassle to not have to maintain duplicate grammars formats for every single language.

**Comment by schrieveslaach** [comment]
Yes, would be really nice, especially for `bat` (see https://github.com/sharkdp/bat/issues/2129)

**Comment by keith-hall** [comment]
Pretty sure ST has no problem with `\G` - the old Markdown syntax used to be littered with them. But yes, any `tmLanguage` features introduced in the last few years are likely unsupported by ST. It doesn't mean `syntect` can't support them though.

**Comment by yankeeinlondon** [comment]
Just wondering ... is this just a request or is there active development on this? Seems like a really powerful upgrade and am I right that support for Typescript is dependant on this?

**Comment by keith-hall** [comment]
No, Typescript support is dependant on https://github.com/trishume/syntect/issues/271 and https://github.com/trishume/syntect/issues/323

**Comment by acao** [comment]
> VSCode has things like injections.

@michaelblyons indeed, as well as atom, I would _love_ to use this crate and dify or Pixelmatch to build a snapshot regression testing suite for our vscode-ified texmate grammars, but we make heavy use of `injectionSelector`, in some cases several layers deep (e.g. markdown -> python -> graphql). we might still use this just for the base grammar used for all graphql strings and files (discussion: https://github.com/graphql/graphiql/pull/3273)

if anyone else would like to have advanced textmate features like [ `injectionSelector`](https://gist.github.com/Aerijo/b8c82d647db783187804e86fa0a604a1#injections) support, I'd be happy to collaborate! I think it's possible that `injectionSelector` configs could be transformed into ST `include` configs at runtime if those are supported?

**Comment by CdRGit** [comment]
is there any update on this? I want to add a custom syntax to one of my projects, and a vscode syntax exists for it but I cannot for the life of me find any way to automatically convert it to sublime-syntax? if I'm missing something there and someone knows where to find that in a way that doesn't rely on an unsupported vscode extension (I really don't want to install vscode just to convert a single file) that'd be nice to hear too

**Comment by keith-hall** [comment]
A VSCode syntax as in some JSONified tmLanguage? Maybe you can find a converter to get it back into PList/XML format and then work with it from there. Maybe the Python plugin Sublime Text comes with can be adapted to work without Sublime Text.
